### PR TITLE
Take snapshots of hat map

### PR DESCRIPTION
**Pull request by pokey**
If the user has latest vscode client installed, we use hat snapshots to keep hats from changing during a phrase.  If not we fall back to no snapshots, like we had before

### CODE REVIEW

File: src/command.py [hunk 1]
```diff
@@ -78,15 +78,33 @@ def cursorless_multiple_target_command(
...
+        use_pre_phrase_snapshot = actions.user.did_emit_pre_phrase_signal()
+    except KeyError:
+        use_pre_phrase_snapshot = False
+
+    return {
+        "version": 1,
+        "spokenForm": get_spoken_form(),
+        "action": action,
+        "targets": targets,
+        "extraArgs": args,
```
Comment (AndreasArvidsson) [inline src/command.py:104]: extraArgs I'm not sold on this name. May be just `arguments`?

**Comment by AndreasArvidsson** [comment]
Looks good overall. Just realized it looks like we have a duplication of `cursorless_single_target_command_with_arg_list()`. Could you remove one of these?

File: src/command.py (hunk 1 above)
Comment (pokey) [reply src/command.py:104]: Yeah thought a bit about this one. They're "extra" in the sense that targets are also arguments to the action. Something does smell a bit weird here. But I think it's prob a symptom of the fact that targets have such special treatment relative to other args to the action. Idk

Comment (AndreasArvidsson) [reply src/command.py:104]: I get what you're coming from. I can live with extraArgs

**Comment by pokey** [comment]
> Looks good overall. Just realized it looks like we have a duplication of `cursorless_single_target_command_with_arg_list()`. Could you remove one of these?

I think I actually have that removed on a different PR branch, but I'll have a look

**Review by AndreasArvidsson — APPROVED** [review]

**Pull request merged**